### PR TITLE
[NEXMANAGE-1101] Fix history log issue

### PR DESCRIPTION
**Pull request by yengliong93**
### PULL DESCRIPTION

INBM checks the start time of the history log entries. If the log entries are within the OTA timeframe, it will proceed to parse the log and save the package-level information into the granular log file.

### Impact Analysis

| Info | Please fill out this column |
| ------ | ----------- |
| Root Cause | When INBM looks at the history.log, it always picks the last upgrade log entries. To capture all the upgrade information, INBM needs a logic to parse all the upgrade data within the desired timeframe. |
| Jira ticket | NEXMANAGE-1101 |

### CODE MAINTAINABILITY

- [ ] Added required new tests relevant to the changes
- [ ] Updated Documentation as relevant to the changes
- [ ] PR change contains code related to security
- [ ] PR introduces changes that break compatibility with other modules/services (If YES, please provide description)
- [ ] Run `go fmt` or `format-python.sh` as applicable
- [x] Update Changelog
- [ ] Integration tests are passing
- [ ] If Cloudadapter changes, check Azure connectivity manually

# _Code must act as a teacher for future developers_

### CODE REVIEW

**Comment by yengliong93** [comment]
@gblewis1 please have a review, thanks.

File: inbm/dispatcher-agent/dispatcher/update_logger.py [hunk 1]
```diff
@@ -140,40 +140,47 @@ def update_granular_with_sota_package_upgrade(self) -> None:
...
-                    data['UpdateLog'].append(package_info)
-
-                # Open the file in write mode to save the updated data
-                with open(GRANULAR_LOG_FILE, 'w') as f:
-                    json.dump(data, f, indent=4)
+                    upgrade_parts_shadow = upgrade_parts.copy()
+                    # Only look at entries >= SOTA time, removing the old entries
+                    for upgrade_part in upgrade_parts_shadow:
+                        upgrade_time = get_package_start_date(upgrade_part)
+                        if datetime.datetime.fromisoformat(upgrade_time) < self._time:
```
Comment (gblewis1) [inline inbm/dispatcher-agent/dispatcher/update_logger.py:147]: (removed comment, see below)

Comment (gblewis1) [reply inbm/dispatcher-agent/dispatcher/update_logger.py:147]: i would suggest passing the update start time in as keep track of time falls outside the reasonable scope of a logger; dispatcher itself should be keeping track of this time

Comment (gblewis1) [reply inbm/dispatcher-agent/dispatcher/update_logger.py:147]: ah, i see that the `_time` is set again when the OTA starts

**Review by gblewis1 — APPROVED** [review]

**Pull request merged**